core/consensus/beefy/impl/beefy_impl.hpp

```diff
@@ -161,6 +169,7 @@ namespace kagome::network {
     primitives::events::ChainSub chain_sub_;
 
     std::optional<primitives::BlockNumber> beefy_genesis_;
+    std::optional<FetchingHeaders> fetching_headers_;
```

Comment:
`finalizationEstablished` not needed, because beefy only works with finalized blocks
```suggestion
    std::optional<primitives::BlockInfo> fetching_headers_;
```

Reply:
We should know somehow that finalization of block sequence was established. Theoretically we can start fetching headers not from grandpa finalized but from some other block, that might be not finalized.

Reply:
No,
You are confusing `Synchronizer::fetchHeadersBack` which can be called with any block,
and `BeefyImpl::fetching_headers_` which is initialized with `grandpa_finalized`.
Beefy only works with finalized blocks.